app/models/CreditDetailModel.scala

```diff
@@ -17,14 +17,21 @@
 package models
 
 import models.financialDetails.{DocumentDetail, FinancialDetailsModel}
-import models.paymentAllocationCharges.FinancialDetailsWithDocumentDetailsModel
+
 
 import java.time.LocalDate
 
 case class CreditDetailModel(date: LocalDate , documentDetail: DocumentDetail, creditType: CreditType)
 
-sealed trait CreditType
+sealed trait CreditType {
+  val creditTypeAsString: String
+}
+
+case object MfaCreditType extends CreditType {
+  override val creditTypeAsString = "Credit from HMRC adjustment"
```

Comment:
we should use language resources where required by the context

Reply:
not sure how to implement this? We don't have access to messages in CreditDetailModel.scala

Reply:
I'll replace the text with the messages key